crates/starknet-devnet/tests/common/background_devnet.rs

```diff
@@ -167,15 +168,27 @@ impl BackgroundDevnet {
         &self,
         method: &str,
         params: serde_json::Value,
-    ) -> serde_json::Value {
+    ) -> Result<serde_json::Value, RpcError> {
         let body_json = json!({
             "jsonrpc": "2.0",
             "id": 0,
             "method": method,
             "params": params
         });
 
-        self.reqwest_client().post_json_async(RPC_PATH, body_json).await.unwrap()
+        let json_rpc_result: serde_json::Value = self
+            .reqwest_client()
+            .post_json_async(RPC_PATH, body_json)
+            .await
+            .map_err(|_| RpcError::internal_error())?;
+
+        if let Some(result) = json_rpc_result.get("result") {
+            Ok(result.clone())
+        } else if let Some(error) = json_rpc_result.get("error") {
+            Err(serde_json::from_value(error.clone()).unwrap())
+        } else {
+            Err(RpcError::internal_error())
```

Comment:
Isn't this hiding the real response we are getting? Is there an `RpcError` variant which can wrap the response, at least in the stringified form?

Reply:
the real response is in the "error" entry of the json response. The internal_error indicates that this json response is malformed

Reply:
I am aware of the case when the real error is in a dedicated property. But this way we'll have the same behavior if the server responds with 500, 400 or returns an invalid response. It is testing code, so we won't be affecting users, it just might be harder for us to debug our tests in the future. And maybe these "hidden error" scenarios are not even possible, I don't know.

Reply:
[46fb94a](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/500/commits/46fb94aada82088d4c08ca6c1105ac9b5f22788b)
[a7193a0](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/500/commits/a7193a08df3233f4ea15a5158d071981bb79b052)